torch/tensor.py

```diff
@@ -103,7 +103,14 @@ def __deepcopy__(self, _memo):
         return new_tensor
 
     def __reduce__(self):
-        return type(self), (self.tolist(),)
+        return (type(self), (),
+                (self.storage(),
+                self.storage_offset(),
+                tuple(self.size()),
+                self.stride()))
+
+    def __setstate__(self, state):
```

Comment:
Could you implement `__getstate__` too? Just to be safe, idk if some functions won't start using the new method now.